src/spec-node/dockerCompose.ts

```diff
@@ -327,32 +330,18 @@ async function startContainer(params: DockerResolverParameters, buildParams: Doc
 				composeGlobalArgs.push('-f', persistedStartFile);
 			}
 		}
-		if (!didRestoreFromPersistedShare) {
-			const imageName = container.Config.Image;
-			let cache: Promise<ImageDetails> | undefined;
-			const imageDetails = () => cache || (cache = inspectDockerImage(params, imageName, false));
-			const labelDetails = async () => { return { definition: undefined, version: undefined }; };
-			const featuresConfig = await generateFeaturesConfig(params.common, (await createFeaturesTempFolder(params.common)), config, labelDetails, getContainerFeaturesFolder);
-			if (featuresConfig) {
-				const collapsedFeaturesConfig = collapseFeaturesConfig(featuresConfig);
-				// Save override docker-compose file to disk.
-				// Persisted folder is a path that will be maintained between sessions
-				// Note: As a fallback, persistedFolder is set to the build's tmpDir() directory
-				const overrideFilePath = await writeFeaturesComposeOverrideFile(container.Config.Image, originalImageName, collapsedFeaturesConfig, config, buildParams, composeFiles, imageDetails, service, idLabels, params.additionalMounts, persistedFolder, featuresStartOverrideFilePrefix, buildCLIHost, output);
-				if (overrideFilePath) {
-					// Add file path to override file as parameter
-					composeGlobalArgs.push('-f', overrideFilePath);
-				}
-			}
-		}
-	} else {
+	}
+
+	if (!container || !didRestoreFromPersistedShare) {
```

Comment:
There is a case where we have a "build" but no "start" file. In that case `if (persistedBuildFile && persistedStartFile) {` would not hold and `didRestoreFromPersistedShare` would be false.